pkg/csi-handler/pv_checker.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi_handler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"k8s.io/klog"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/kubernetes-csi/external-health-monitor/pkg/util"
+)
+
+// PVHealthConditionChecker is for checking pv health condition
+type PVHealthConditionChecker struct {
+	checkerName string
+
+	csiConn       *grpc.ClientConn
+	timeout       time.Duration
+	k8sClient     kubernetes.Interface
+	eventRecorder record.EventRecorder
+
+	pvcLister corelisters.PersistentVolumeClaimLister
+	pvLister  corelisters.PersistentVolumeLister
+
+	csiPVHandler CSIHandler
+}
+
+// NewPVHealthConditionChecker return an instance of PVHealthConditionChecker
+func NewPVHealthConditionChecker(name string, conn *grpc.ClientConn, kClient kubernetes.Interface, timeout time.Duration,
+	pvcLister corelisters.PersistentVolumeClaimLister, pvLister corelisters.PersistentVolumeLister, recorder record.EventRecorder) *PVHealthConditionChecker {
+
+	return &PVHealthConditionChecker{
+		checkerName:   name,
+		csiConn:       conn,
+		k8sClient:     kClient,
+		eventRecorder: recorder,
+		pvcLister:     pvcLister,
+		pvLister:      pvLister,
+		timeout:       timeout,
+
+		csiPVHandler: NewCSIPVHandler(conn),
+	}
+}
+
+// CheckControllerVolumesStatus checks volumes health condition by ListVolumes
+func (checker *PVHealthConditionChecker) CheckControllerVolumesStatus() error {
```

Comment:
The function names CheckControllerVolumesStatus and CheckControllerVolumeStatus are almost identical so it is confusing.  Can you rename this one CheckControllerVolumesStatus to CheckControllerListVolumeStatuses?